src/figuresClasses.ts

```diff
@@ -1,19 +1,89 @@
 export interface Figure {
-
+  shape: 'triangle' | 'circle' | 'rectangle',
+  color: 'red' | 'green' | 'blue',
+  area: number
 }
 
 export class Triangle implements Figure {
+  public shape: Figure['shape'];
+
+  public area: number;
+
+  constructor(
+    public color: Figure['color'],
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    this.shape = 'triangle';
+    this.area = this.getArea();
+  }
+
+  getArea(): number {
+    const { a, b, c } = this;
+
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw Error('Wrong data');
```

Comment:
[CODE STYLE] - If you throw an error try to provide its detailed description in error message.

check this for all code